modules/datahandling/filerequests.go

```diff
@@ -0,0 +1,143 @@
+package datahandling
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+var fileRequestsSetup = false
+
+// initProjectRequests populates the requestMap from requestmap.go with the appropriate constructors for the project methods
+func initFileRequests() {
+	if fileRequestsSetup {
+		return
+	}
+
+	authenticatedRequestMap["FileCreate"] = func(req abstractRequest) (request, error) {
+		p := new(fileCreateRequest)
+		p.abstractRequest = req
+		rawData := p.Data
+		err := json.Unmarshal(rawData, &p)
+		return p, err
+	}
+
+	authenticatedRequestMap["FileRename"] = func(req abstractRequest) (request, error) {
+		p := new(fileRenameRequest)
+		p.abstractRequest = req
+		rawData := p.Data
+		err := json.Unmarshal(rawData, &p)
+		return p, err
+	}
+
+	authenticatedRequestMap["FileMove"] = func(req abstractRequest) (request, error) {
+		p := new(fileMoveRequest)
+		p.abstractRequest = req
+		rawData := p.Data
+		err := json.Unmarshal(rawData, &p)
+		return p, err
+	}
+
+	authenticatedRequestMap["FileDelete"] = func(req abstractRequest) (request, error) {
+		p := new(fileDeleteRequest)
+		p.abstractRequest = req
+		rawData := p.Data
+		err := json.Unmarshal(rawData, &p)
+		return p, err
+	}
+
+	authenticatedRequestMap["FileChange"] = func(req abstractRequest) (request, error) {
+		p := new(fileChangeRequest)
+		p.abstractRequest = req
+		rawData := p.Data
+		err := json.Unmarshal(rawData, &p)
+		return p, err
+	}
+
+	authenticatedRequestMap["FilePull"] = func(req abstractRequest) (request, error) {
+		p := new(filePullRequest)
+		p.abstractRequest = req
+		rawData := p.Data
+		err := json.Unmarshal(rawData, &p)
+		return p, err
+	}
```

Comment:
Why are these not a helper function? You literally have the same code block 5 times.